torch/_subclasses/meta_utils.py

```diff
@@ -160,7 +161,13 @@ def meta_storage(self, s, callback):
     # as part of this process, we will maintain this invariant!  (Even though
     # other users of this may not need it this property to be upheld.)
     def meta_tensor(
-        self, t, shape_env=None, callback=lambda t: t(), source: Optional[Source] = None
+        self,
+        t,
+        shape_env=None,
+        callback=lambda t: t(),
+        source: Optional[Source] = None,
+        dynamic_dims: Optional[List[DIM_DYNAMISM_STATE]] = None,
+        dynamic_dims_range: Optional[Dict[int, MinMaxConstraint]] = None,
```

Comment:
Naming this something more generic (e.g., constraints) will save you from having to do a mass rename (and change the type annotation MinMaxConstraint) later, if other constraints show up.

Reply:
I dislike generic naming for future refactors, because I do not know when those will show up, if ever, but I do not have strong feelings.

Reply:
My point is that the high level export API is generically named (constraints). If you think that we will never have a constraint besides min-max, then you should lobby harder to make the high level API less generically named too. But I think everyone agrees that it's fine to allow for this possibility. Then I would advocate for consistency between the user facing API and the internal mechanism.

Reply:
Sold, can rename.